python/tests/test_tables.py

```diff
@@ -1009,7 +1010,162 @@ def test_set_with_optional_properties(self, codec):
                 assert md == row.metadata
 
 
-class TestIndividualTable(CommonTestsMixin, MetadataTestsMixin):
+class AssertEqualsMixin:
+    @pytest.fixture
+    def test_rows(self, scope="session"):
+        test_rows = self.make_transposed_input_data(10)
+        # Annoyingly we have to tweak some types as once added to a row and then put in
+        # an error message things come out differently
+        for n in range(10):
+            for col in test_rows[n].keys():
+                if col in ["timestamp", "record", "ancestral_state", "derived_state"]:
+                    test_rows[n][col] = bytes(test_rows[n][col]).decode("ascii")
+        return test_rows
+
+    @pytest.fixture
+    def table1(self, test_rows):
+        table1 = self.table_class()
+        for row in test_rows[:5]:
+            table1.add_row(**row)
+        return table1
+
+    def test_equal(self, table1, test_rows):
+        table2 = self.table_class()
+        for row in test_rows[:5]:
```

Comment:
isn't this depending on the number of rows being 5? e.g., shouldn't you replace `5` with `table1.num_rows`?

Reply:
The tables have a fix number of rows (10).